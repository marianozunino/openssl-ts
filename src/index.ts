import { spawn } from 'child_process';
import util from 'util';

const debuglog = util.debuglog('openssl');

const executeCommand = (
  command: string,
  flags: string[],
  stdinContent?: Buffer,
): Buffer | PromiseLike<Buffer> => {
  return new Promise<Buffer>((resolve, reject) => {
    const output: Buffer[] = [];
    debuglog(`Executing command: ${command} ${flags.join(' ')}`);
    const childProcess = spawn(command, flags);

    if (stdinContent) {
      debuglog(`Sending stdin content: ${stdinContent.toString()}`);
      childProcess.stdin.cork();
      childProcess.stdin.write(stdinContent);
      childProcess.stdin.uncork();
    }

    childProcess.stderr.on('data', (data: Buffer) => {
      output.push(data);
    });

    childProcess.stdout.on('data', (data: Buffer) => {
      output.push(data);
    });

    childProcess.on('error', (err) => {
      return reject(err);
    });

    childProcess.on('close', (code) => {
      if (code !== 0) {
        return reject(
          new Error(
            `openssl exited with code ${code}.\n${Buffer.concat(
              output,
            ).toString()}`,
          ),
        );
      } else {
        return resolve(Buffer.concat(output));
      }
    });
  });
};

export const openssl = async (
  flags: Array<string>,
  options?: {
    openSSLPath?: string;
    stdin?: Buffer;
  },
): Promise<Buffer> => {
  if (!Array.isArray(flags)) {
    throw new Error("'flags' must be a string array");
  }

  flags.forEach((flag) => {
    if (typeof flag !== 'string') {
      throw new Error(`'flags' must be a string array`);
    }
  });

  if (flags.length === 0) {
    throw new Error("'flags' must not be empty");
  }

  const opensslBin =
    options?.openSSLPath || process.env.OPENSSL_PATH || 'openssl';

  return executeCommand(opensslBin, flags, options?.stdin);
};
