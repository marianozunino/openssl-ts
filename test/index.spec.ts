import { existsSync, readFileSync, unlinkSync } from 'fs';
import { tmpdir } from 'os';
import { join } from 'path';
import { openssl } from 'src/index';

describe('openssl', () => {
  const outputFile = join(tmpdir(), 'output.key');
  beforeEach(() => {
    delete process.env.OPENSSL_PATH;
    if (existsSync(outputFile)) {
      unlinkSync(outputFile);
    }
  });

  describe('when invalid flags are used', () => {
    it('should throw if flags are empty', async () => {
      await expect(openssl([])).rejects.toThrowErrorMatchingSnapshot();
    });

    it('should throw if flags are not string', async () => {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      await expect(openssl([1])).rejects.toThrowErrorMatchingSnapshot();
    });

    it('should throw if flags are not string', async () => {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      await expect(openssl('genrsa')).rejects.toThrowErrorMatchingSnapshot();
    });
  });

  describe('when openssl binary is not found', () => {
    it('should throw if openSSLPath is invalid', async () => {
      await expect(
        openssl(['someflag'], {
          openSSLPath: 'openSSLPath-not-a-real-openssl-binary',
        }),
      ).rejects.toThrowErrorMatchingSnapshot();
    });

    it('should throw if OPENSSL_PATH env is invalid', async () => {
      process.env.OPENSSL_PATH = 'OPENSSL_PATH-not-a-real-openssl-binary';
      await expect(
        openssl(['someflag']),
      ).rejects.toThrowErrorMatchingSnapshot();
    });
  });

  describe('should execute openssl without errors', () => {
    it('should execute openssl without errors', async () => {
      const result = await openssl(['exit'], {
        openSSLPath: 'openssl',
      });
      expect(result).toBeInstanceOf(Buffer);
      expect(result.toString()).toMatchSnapshot();
    });
    describe('genrsa', () => {
      it('should support genrsa action as stdout output', async () => {
        const pass = '123456';
        const rsa = await openssl([
          'genrsa',
          '-des3',
          '-passout',
          `pass:${pass}`,
          '1048',
        ]);
        expect(
          rsa
            .toString()
            .includes(
              'Generating RSA private key, 1048 bit long modulus (2 primes)',
            ),
        ).toBeTruthy();
        expect(
          rsa.toString().includes('-----BEGIN RSA PRIVATE KEY-----'),
        ).toBeTruthy();
        expect(
          rsa.toString().includes('-----END RSA PRIVATE KEY-----'),
        ).toBeTruthy();
      });
      it('should support genrsa action as stdout output', async () => {
        const pass = '123456';
        const rsa = await openssl([
          'genrsa',
          '-des3',
          '-out',
          outputFile,
          '-passout',
          `pass:${pass}`,
          '1048',
        ]);
        expect(
          rsa
            .toString()
            .includes(
              'Generating RSA private key, 1048 bit long modulus (2 primes)',
            ),
        ).toBeTruthy();
        expect(
          rsa.toString().includes('-----BEGIN RSA PRIVATE KEY-----'),
        ).toBeFalsy();
        expect(
          rsa.toString().includes('-----END RSA PRIVATE KEY-----'),
        ).toBeFalsy();
        expect(existsSync(outputFile)).toBeTruthy();
        const content = readFileSync(outputFile, 'utf8');
        expect(
          content.startsWith('-----BEGIN RSA PRIVATE KEY-----'),
        ).toBeTruthy();
        expect(content.includes('-----END RSA PRIVATE KEY-----')).toBeTruthy();
      });
    });
    describe('smime verify', () => {
      it('should support smime verify action', async () => {
        const output = await openssl([
          'smime',
          '-verify',
          '-inform',
          'DER',
          '-in',
          `${__dirname}/__resources__/signed.mobileprovision`,
          '-noverify',
        ]);
        expect(output.toString()).toMatchSnapshot();
      });
      it('should support smime verify action using stdin', async () => {
        const buffer = readFileSync(
          `${__dirname}/__resources__/signed.mobileprovision`,
        );
        const output = await openssl(
          ['smime', '-verify', '-inform', 'DER', '-noverify'],
          { stdin: buffer },
        );
        expect(output.toString()).toMatchSnapshot();
      });
    });
    describe('x509', () => {
      it('should support x509 action', async () => {
        const output = await openssl([
          'x509',
          '-in',
          `${__dirname}/__resources__/AppleIncRootCertificate.cer`,
          '-subject',
          '-noout',
          '-inform',
          'DER',
        ]);
        expect(output.toString()).toMatchSnapshot();
      });
      it('should support x509 action using stdin', async () => {
        const buffer = readFileSync(
          `${__dirname}/__resources__/AppleIncRootCertificate.cer`,
        );
        const output = await openssl(
          ['x509', '-subject', '-noout', '-inform', 'DER'],
          { stdin: buffer },
        );
        expect(output.toString()).toMatchSnapshot();
      });
    });
  });

  describe('should support cms action', () => {
    it('without stdin', async () => {
      const output = await openssl([
        'cms',
        '-sign',
        '-signer',
        `${__dirname}/__resources__/domain.crt`,
        '-inkey',
        `${__dirname}/__resources__/domain.key`,
        '-nodetach',
        '-outform',
        'der',
        '-in',
        `${__dirname}/__resources__/signme.xml`,
      ]);
      expect(
        output
          .toString('base64')
          .startsWith(
            'MIIFyQYJKoZIhvcNAQcCoIIFujCCBbYCAQExDTALBglghkgBZQMEAgEwggE5BgkqhkiG9w0BBwGgggEqBIIBJjxsb2dpblRpY2tldFJlcXVlc3QgdmVyc2lvbj0iMS4wIj4NCiAgPGhlYWRlcj4NCiAgICA8dW5pcXVlSWQ+MTY0NjMxNjI4ODwvdW5pcXVlSWQ+DQogICAgPGdlbmVyYXRpb25UaW1lPjIwMjItMDMtMDNUMTA6MzQ6NDguODUzLTAzOjAwPC9nZW5lcmF0aW9uVGltZT4NCiAgICA8ZXhwaXJhdGlvblRpbWU+MjAyMi0wMy0wM1QxMTozNDo0OC44NTUtMDM6MDA8L2V4cGlyYXRpb25UaW1lPg0KICA8L2hlYWRlcj4NCiAgPHNlcnZpY2U+d3Nfc3JfcGFkcm9uX2ExMzwvc2VydmljZT4NCjwvbG9naW5UaWNrZXRSZXF1ZXN0PqCCAnAwggJsMIIB0qADAgECAhRm9+mmWM7kzqxuMMl+xA8G7/KBBjANBgkqhkiG9w0BAQsFADBFMQswCQYDVQQGEwJBVTETMBEGA1UECAwKU29tZS1TdGF0ZTEhMB8GA1UECgwYSW50ZXJuZXQgV2lkZ2l0cyBQdHkgTHRkMB4XDTIyMDMwMzE0MzQ1NFoXDTIzMDMwMzE0MzQ1NFowRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoMGEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDCBojANBgkqhkiG9w0BAQEFAAOBkAAwgYwCgYQAzF0Hw/TnOlgsYzbR3iOMFgCPnQ6h0e85E21PXtNzMS//6Yez22fg4jI4GAj5ho2M7YjHz0BvVyxbp4pT4riqT2LnSb0x1wNVfHymdx7j1ygq7AlpZYc0pw+',
          ),
      ).toBeTruthy();
    });
    it('with stdin', async () => {
      const stdin = readFileSync(`${__dirname}/__resources__/signme.xml`);
      const output = await openssl(
        [
          'cms',
          '-sign',
          '-signer',
          `${__dirname}/__resources__/domain.crt`,
          '-inkey',
          `${__dirname}/__resources__/domain.key`,
          '-nodetach',
          '-outform',
          'der',
        ],
        {
          stdin,
        },
      );
      expect(
        output
          .toString('base64')
          .startsWith(
            'MIIFyQYJKoZIhvcNAQcCoIIFujCCBbYCAQExDTALBglghkgBZQMEAgEwggE5BgkqhkiG9w0BBwGgggEqBIIBJjxsb2dpblRpY2tldFJlcXVlc3QgdmVyc2lvbj0iMS4wIj4NCiAgPGhlYWRlcj4NCiAgICA8dW5pcXVlSWQ+MTY0NjMxNjI4ODwvdW5pcXVlSWQ+DQogICAgPGdlbmVyYXRpb25UaW1lPjIwMjItMDMtMDNUMTA6MzQ6NDguODUzLTAzOjAwPC9nZW5lcmF0aW9uVGltZT4NCiAgICA8ZXhwaXJhdGlvblRpbWU+MjAyMi0wMy0wM1QxMTozNDo0OC44NTUtMDM6MDA8L2V4cGlyYXRpb25UaW1lPg0KICA8L2hlYWRlcj4NCiAgPHNlcnZpY2U+d3Nfc3JfcGFkcm9uX2ExMzwvc2VydmljZT4NCjwvbG9naW5UaWNrZXRSZXF1ZXN0PqCCAnAwggJsMIIB0qADAgECAhRm9+mmWM7kzqxuMMl+xA8G7/KBBjANBgkqhkiG9w0BAQsFADBFMQswCQYDVQQGEwJBVTETMBEGA1UECAwKU29tZS1TdGF0ZTEhMB8GA1UECgwYSW50ZXJuZXQgV2lkZ2l0cyBQdHkgTHRkMB4XDTIyMDMwMzE0MzQ1NFoXDTIzMDMwMzE0MzQ1NFowRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoMGEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDCBojANBgkqhkiG9w0BAQEFAAOBkAAwgYwCgYQAzF0Hw/TnOlgsYzbR3iOMFgCPnQ6h0e85E21PXtNzMS//6Yez22fg4jI4GAj5ho2M7YjHz0BvVyxbp4pT4riqT2LnSb0x1wNVfHymdx7j1ygq7AlpZYc0pw+',
          ),
      ).toBeTruthy();
    });
  });
});
