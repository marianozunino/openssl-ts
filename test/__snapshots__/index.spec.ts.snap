// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`openssl should execute openssl without errors smime verify should support smime verify action 1`] = `
"Verification successful
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>AppIDName</key>
	<string>LesDocksMarseille</string>
	<key>ApplicationIdentifierPrefix</key>
	<array>
	<string>B7U2JQ28D6</string>
	</array>
	<key>CreationDate</key>
	<date>2016-05-02T09:13:29Z</date>
	<key>Platform</key>
	<array>
		<string>iOS</string>
	</array>
	<key>DeveloperCertificates</key>
	<array>
		<data>MIIFoTCCBImgAwIBAgIIadMmad9a8Q8wDQYJKoZIhvcNAQEFBQAwgZYxCzAJBgNVBAYTAlVTMRMwEQYDVQQKDApBcHBsZSBJbmMuMSwwKgYDVQQLDCNBcHBsZSBXb3JsZHdpZGUgRGV2ZWxvcGVyIFJlbGF0aW9uczFEMEIGA1UEAww7QXBwbGUgV29ybGR3aWRlIERldmVsb3BlciBSZWxhdGlvbnMgQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkwHhcNMTYwNTAyMDgzNzI0WhcNMTcwNTAyMDgzNzI0WjCBlDEaMBgGCgmSJomT8ixkAQEMCkI3VTJKUTI4RDYxOjA4BgNVBAMMMWlQaG9uZSBEaXN0cmlidXRpb246IENhcmxpcGEgU3lzdGVtcyAoQjdVMkpRMjhENikxEzARBgNVBAsMCkI3VTJKUTI4RDYxGDAWBgNVBAoMD0NhcmxpcGEgU3lzdGVtczELMAkGA1UEBhMCVVMwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDMv9H6gI3Wd0IWovykpvFMSuySp8T0askMS3AfNlQLN7n7Q3J5ptyZ/OvH4rJcd7t4HOuy6Nac6vY0pIMmJsTc470X51ThTshTHcfoaA8MIYvuO3SDFJZBAmv84Yryzk91CrKUTucJmums2m7R+r0ial9fAitBQqNi5oYlaUlC9YzDv+tPwa20mhYC81Orhlh/ZKuNml73UCkrB2G1vunFoympnpCDfPR+OGz9WKQeyJpeuiIbzjJ0hmqqDhsbGfQYkeiW9i6XxQ1+y7wxXHfxfC8Uh+JvLMZ5X77LKmwkGmWLT3J6DBNqtBEEl7wn7eiJtLOLSqdQm+Qa3DbJxL9NAgMBAAGjggHxMIIB7TAdBgNVHQ4EFgQUmsLaBz9jQUGzXOrZHeqsXQbxqgswDAYDVR0TAQH/BAIwADAfBgNVHSMEGDAWgBSIJxcJqbYYYIvs67r2R1nFUlSjtzCCAQ8GA1UdIASCAQYwggECMIH/BgkqhkiG92NkBQEwgfEwgcMGCCsGAQUFBwICMIG2DIGzUmVsaWFuY2Ugb24gdGhpcyBjZXJ0aWZpY2F0ZSBieSBhbnkgcGFydHkgYXNzdW1lcyBhY2NlcHRhbmNlIG9mIHRoZSB0aGVuIGFwcGxpY2FibGUgc3RhbmRhcmQgdGVybXMgYW5kIGNvbmRpdGlvbnMgb2YgdXNlLCBjZXJ0aWZpY2F0ZSBwb2xpY3kgYW5kIGNlcnRpZmljYXRpb24gcHJhY3RpY2Ugc3RhdGVtZW50cy4wKQYIKwYBBQUHAgEWHWh0dHA6Ly93d3cuYXBwbGUuY29tL2FwcGxlY2EvME0GA1UdHwRGMEQwQqBAoD6GPGh0dHA6Ly9kZXZlbG9wZXIuYXBwbGUuY29tL2NlcnRpZmljYXRpb25hdXRob3JpdHkvd3dkcmNhLmNybDAOBgNVHQ8BAf8EBAMCB4AwFgYDVR0lAQH/BAwwCgYIKwYBBQUHAwMwEwYKKoZIhvdjZAYBBAEB/wQCBQAwDQYJKoZIhvcNAQEFBQADggEBAI+5Whb0zG+eeuajCnPazPtY0bcODEG4SBjWgedcv1aSHlaZ1dELdWtZXoaRfKnrOLu/duY8c3yjidVKNlMS/+stIq+BDgyc62uZhb5O1Z3MHtHCqvixutXwyzXIF+Lw6c/f1wRn/JYK6rIzxTWGGnb25sHaSscUGcL/VYm86I4tkACUCmk/qwTTJNb9PzYvubVtNv72XqT4KhgP1SGzJD9N8daID3kmctuNLdHe0CAottPuEbLOB59A/qfSrE4zExjYw66L235pmxNHE58u8Aecwkn+jzOmYwcgA6dd8aLlo0vLeIaXVjzMicM3Pn4XoIJGGdY2VuiaxL6cAIqmsW4=</data>
	</array>
	<key>Entitlements</key>
	<dict>
		<key>keychain-access-groups</key>
		<array>
			<string>B7U2JQ28D6.*</string>		
		</array>
		<key>get-task-allow</key>
		<false/>
		<key>application-identifier</key>
		<string>B7U2JQ28D6.com.carlipa.LesDocksMarseille</string>
		<key>com.apple.developer.team-identifier</key>
		<string>B7U2JQ28D6</string>
		<key>beta-reports-active</key>
		<true/>
	</dict>
	<key>ExpirationDate</key>
	<date>2017-05-02T08:37:24Z</date>
	<key>Name</key>
	<string>foo</string>
	<key>TeamIdentifier</key>
	<array>
		<string>B7U2JQ28D6</string>
	</array>
	<key>TeamName</key>
	<string>Carlipa Systems</string>
	<key>TimeToLive</key>
	<integer>364</integer>
	<key>UUID</key>
	<string>a75f9275-e719-4af1-be35-b80255ab9c53</string>
	<key>Version</key>
	<integer>1</integer>
</dict>
</plist>"
`;

exports[`openssl should execute openssl without errors smime verify should support smime verify action using stdin 1`] = `
"Verification successful
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>AppIDName</key>
	<string>LesDocksMarseille</string>
	<key>ApplicationIdentifierPrefix</key>
	<array>
	<string>B7U2JQ28D6</string>
	</array>
	<key>CreationDate</key>
	<date>2016-05-02T09:13:29Z</date>
	<key>Platform</key>
	<array>
		<string>iOS</string>
	</array>
	<key>DeveloperCertificates</key>
	<array>
		<data>MIIFoTCCBImgAwIBAgIIadMmad9a8Q8wDQYJKoZIhvcNAQEFBQAwgZYxCzAJBgNVBAYTAlVTMRMwEQYDVQQKDApBcHBsZSBJbmMuMSwwKgYDVQQLDCNBcHBsZSBXb3JsZHdpZGUgRGV2ZWxvcGVyIFJlbGF0aW9uczFEMEIGA1UEAww7QXBwbGUgV29ybGR3aWRlIERldmVsb3BlciBSZWxhdGlvbnMgQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkwHhcNMTYwNTAyMDgzNzI0WhcNMTcwNTAyMDgzNzI0WjCBlDEaMBgGCgmSJomT8ixkAQEMCkI3VTJKUTI4RDYxOjA4BgNVBAMMMWlQaG9uZSBEaXN0cmlidXRpb246IENhcmxpcGEgU3lzdGVtcyAoQjdVMkpRMjhENikxEzARBgNVBAsMCkI3VTJKUTI4RDYxGDAWBgNVBAoMD0NhcmxpcGEgU3lzdGVtczELMAkGA1UEBhMCVVMwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDMv9H6gI3Wd0IWovykpvFMSuySp8T0askMS3AfNlQLN7n7Q3J5ptyZ/OvH4rJcd7t4HOuy6Nac6vY0pIMmJsTc470X51ThTshTHcfoaA8MIYvuO3SDFJZBAmv84Yryzk91CrKUTucJmums2m7R+r0ial9fAitBQqNi5oYlaUlC9YzDv+tPwa20mhYC81Orhlh/ZKuNml73UCkrB2G1vunFoympnpCDfPR+OGz9WKQeyJpeuiIbzjJ0hmqqDhsbGfQYkeiW9i6XxQ1+y7wxXHfxfC8Uh+JvLMZ5X77LKmwkGmWLT3J6DBNqtBEEl7wn7eiJtLOLSqdQm+Qa3DbJxL9NAgMBAAGjggHxMIIB7TAdBgNVHQ4EFgQUmsLaBz9jQUGzXOrZHeqsXQbxqgswDAYDVR0TAQH/BAIwADAfBgNVHSMEGDAWgBSIJxcJqbYYYIvs67r2R1nFUlSjtzCCAQ8GA1UdIASCAQYwggECMIH/BgkqhkiG92NkBQEwgfEwgcMGCCsGAQUFBwICMIG2DIGzUmVsaWFuY2Ugb24gdGhpcyBjZXJ0aWZpY2F0ZSBieSBhbnkgcGFydHkgYXNzdW1lcyBhY2NlcHRhbmNlIG9mIHRoZSB0aGVuIGFwcGxpY2FibGUgc3RhbmRhcmQgdGVybXMgYW5kIGNvbmRpdGlvbnMgb2YgdXNlLCBjZXJ0aWZpY2F0ZSBwb2xpY3kgYW5kIGNlcnRpZmljYXRpb24gcHJhY3RpY2Ugc3RhdGVtZW50cy4wKQYIKwYBBQUHAgEWHWh0dHA6Ly93d3cuYXBwbGUuY29tL2FwcGxlY2EvME0GA1UdHwRGMEQwQqBAoD6GPGh0dHA6Ly9kZXZlbG9wZXIuYXBwbGUuY29tL2NlcnRpZmljYXRpb25hdXRob3JpdHkvd3dkcmNhLmNybDAOBgNVHQ8BAf8EBAMCB4AwFgYDVR0lAQH/BAwwCgYIKwYBBQUHAwMwEwYKKoZIhvdjZAYBBAEB/wQCBQAwDQYJKoZIhvcNAQEFBQADggEBAI+5Whb0zG+eeuajCnPazPtY0bcODEG4SBjWgedcv1aSHlaZ1dELdWtZXoaRfKnrOLu/duY8c3yjidVKNlMS/+stIq+BDgyc62uZhb5O1Z3MHtHCqvixutXwyzXIF+Lw6c/f1wRn/JYK6rIzxTWGGnb25sHaSscUGcL/VYm86I4tkACUCmk/qwTTJNb9PzYvubVtNv72XqT4KhgP1SGzJD9N8daID3kmctuNLdHe0CAottPuEbLOB59A/qfSrE4zExjYw66L235pmxNHE58u8Aecwkn+jzOmYwcgA6dd8aLlo0vLeIaXVjzMicM3Pn4XoIJGGdY2VuiaxL6cAIqmsW4=</data>
	</array>
	<key>Entitlements</key>
	<dict>
		<key>keychain-access-groups</key>
		<array>
			<string>B7U2JQ28D6.*</string>		
		</array>
		<key>get-task-allow</key>
		<false/>
		<key>application-identifier</key>
		<string>B7U2JQ28D6.com.carlipa.LesDocksMarseille</string>
		<key>com.apple.developer.team-identifier</key>
		<string>B7U2JQ28D6</string>
		<key>beta-reports-active</key>
		<true/>
	</dict>
	<key>ExpirationDate</key>
	<date>2017-05-02T08:37:24Z</date>
	<key>Name</key>
	<string>foo</string>
	<key>TeamIdentifier</key>
	<array>
		<string>B7U2JQ28D6</string>
	</array>
	<key>TeamName</key>
	<string>Carlipa Systems</string>
	<key>TimeToLive</key>
	<integer>364</integer>
	<key>UUID</key>
	<string>a75f9275-e719-4af1-be35-b80255ab9c53</string>
	<key>Version</key>
	<integer>1</integer>
</dict>
</plist>"
`;

exports[`openssl should execute openssl without errors x509 should support x509 action 1`] = `
"subject=C = US, O = Apple Inc., OU = Apple Certification Authority, CN = Apple Root CA
"
`;

exports[`openssl should execute openssl without errors x509 should support x509 action using stdin 1`] = `
"subject=C = US, O = Apple Inc., OU = Apple Certification Authority, CN = Apple Root CA
"
`;

exports[`openssl when invalid flags are used should throw if flags are empty 1`] = `"'flags' must not be empty"`;

exports[`openssl when invalid flags are used should throw if flags are not string 1`] = `"'flags' must be a string array"`;

exports[`openssl when invalid flags are used should throw if flags are not string 2`] = `"'flags' must be a string array"`;

exports[`openssl when openssl binary is not found should throw if OPENSSL_PATH env is invalid 1`] = `"spawn OPENSSL_PATH-not-a-real-openssl-binary ENOENT"`;

exports[`openssl when openssl binary is not found should throw if openSSLPath is invalid 1`] = `"spawn openSSLPath-not-a-real-openssl-binary ENOENT"`;
